build.gradle

```diff
@@ -105,6 +106,83 @@ tasks.whenTaskAdded {task ->
     }
 }
 
+def splitTestConfig = [
+    dlicDlsflsTest: [
+        description: "Runs Document- and Field-Level Security tests.",
+        filters: [
+            includeTestsMatching: [
+                "org.opensearch.security.dlic.dlsfls.*"
+            ]
+        ]
+    ],
+    dlicRestApiTest: [
+        description: "Runs REST Management API tests.",
+        filters: [
+            includeTestsMatching: [
+                "org.opensearch.security.dlic.rest.*"
+            ]
+        ]
+    ],
+    crossClusterTest: [
+        description: "Runs cross-cluster tests.",
+        filters: [
+            includeTestsMatching: [
+                    "org.opensearch.security.ccstest.*"
+            ]
+        ]
+    ],
+    sslTest: [
+        description: "Runs most of the SSL tests.",
+        filters: [
+            includeTestsMatching: [
+                "org.opensearch.security.ssl.*"
+            ],
+            excludeTestsMatching: [
+                "org.opensearch.security.ssl.OpenSSL*"
+            ]
+        ]
+    ],
+    opensslCiTest: [
+        description: "Runs portion of SSL tests related to OpenSSL. Explained in https://github.com/opensearch-project/security/pull/2301",
+        include: '**/OpenSSL*.class'
+    ],
+    securityIntegrationTest: [
+        description: "Runs integration tests from all classes.",
+        filters: [
+            includeTestsMatching: [
+                "org.opensearch.security.*Integ*"
+            ],
+            excludeTestsMatching: [
+                "org.opensearch.security.sanity.tests.*"
+            ]
+        ]
+    ],
+    indicesTest: [
+        description: "Runs indices tests from all classes.",
+        filters: [
+            includeTestsMatching: [
+                "org.opensearch.security.*indices*"
+            ],
+            excludeTestsMatching: [
+                "org.opensearch.security.sanity.tests.*"
+            ]
+        ]
+    ]
+] as ConfigObject
+
+List<String> taskNames = splitTestConfig.keySet() as List
+
+task listTasksAsJSON {
+    doLast {
+        System.out.println(new JsonBuilder(taskNames))
+    }
+}
+
+task listTasksAsParam {
+    doLast {
+        System.out.println('-x ' + (taskNames).join(' -x '))
+    }
+}
```

Comment:
These seem like hacks, can they be removed?

Reply:
They are  and no, unless someone has better idea how to dynamically generate list of tasks for Github Actions "matrix" strategy [(link to relevant part of ci.yml)](https://github.com/pawel-gudel-eliatra/security/blob/split-ci-tests/.github/workflows/ci.yml#L24)

Why I've decided to use gradle for that:
1. Gradle has knowledge about all tasks. `build.gradle` is source of truth.
2. Generating list of tasks does not need external utilities like: Bash, Python etc.

Reply:
Digging in `listTasksAsParam` doesn't look like it is used, so we can remove?

Do we need the `doLast` block? can the printing on the json string be at the high level e.g. 

```
task listTasksAsJSON {
    System.out.println(new JsonBuilder(taskNames))
}
```

Reply:
> Digging in listTasksAsParam doesn't look like it is used, so we can remove?
Removed.

> Do we need the doLast block?
Without `doLast` output was sometimes mixed with other tasks output. Removed per your request.

Reply:
Restored `doLast` because Github Task were not working properly.

Reply:
Could you add a comment?  IMO this a good case for comments to explain what problem this addresses since its non-obvious

Reply:
Added.